src/common/tests/btree_direct.c

```diff
@@ -496,15 +523,14 @@ sk_btr_kv_operate(enum sk_btr_opc opc, char *str, bool verbose)
 		case BTR_OPC_DELETE_RETAIN:
 			rc = dbtree_delete(sk_toh, &key_iov, &rec_mmid);
 			if (rc != 0) {
-				D_ERROR("Failed to delete %s\n", key);
-				return -1;
+				sprintf(outbuf, "Failed to delete %s\n", key);
```

Comment:
sprintf can be changed to asprintf for printing any data length.

Reply:
I believe asprintf is not suitable for this place.
Reason: we will start having memory leaks since the allocated buffer will not be released due to fail_msg terminating the program.
eg: char *x;
      asprintf(&x, "Test failed\n");
      fail_msg("%s" , &x);  ---> we cannot free x before the fail_msg.

Reply:
That is true. I think that can be the issue with sprintf too, right ?

Reply:
In the present implementation, we are allocating fixed array outbuf[64] which is in stack. sprintf is using this and when program terminates all stack data is given back. asprintf is creating data in heap which needs to be freed before program terminates.

Reply:
ok make sense